lib/dest/write-contents/write-symbolic-link.js

```diff
@@ -1,18 +1,44 @@
 'use strict';
 
-var fs = require('graceful-fs');
+var path = require('path');
 
 var fo = require('../../file-operations');
 
 function writeSymbolicLink(file, optResolver, onWritten) {
-  // TODO handle symlinks properly
-  fs.symlink(file.symlink, file.path, function(symlinkErr) {
-    if (fo.isFatalOverwriteError(symlinkErr)) {
-      return onWritten(symlinkErr);
-    }
-
-    onWritten();
-  });
+  var srcPath = file.symlink;
+
+  var isDirectory = file.isDirectory();
```

Comment:
True, but it would treat correctly it least the cases where it has the necessary information to, or so it seems to me.

It's too bad that node doesn't enforce this for us, but good to have that confirmed at least.

Having read up a little bit on junctions I'm wondering if the current approach, where it'd be the default even, may end up causing more problems than it solves?

As far as I can tell the problem was that symlinking dirs (but not files?) requires admin privileges.

But junctions are dangerous in that they may cause users to lose data (c.f. the caveats in the comments).

Then again that's easy for me to say because I don't use Windows.